js/interface.js

```diff
@@ -2553,10 +2568,23 @@ $('[data-save-rule]').click(function(event) {
 
   $('[data-dismiss="modal"]').click();
 
+  var isAddingRule = $('#configure-rule').find('.modal-title').text() === 'Add new security rule';
+
   if (currentDataSourceRuleIndex === undefined) {
     currentDataSourceRules.push(rule);
+
+    // For Edit security rule adding the rule in current final rules
+    if ((isAddingRule && widgetData.context === 'overlay' && widgetData.tokenId !== selectedTokenId) || !isAddingRule) {
```

Comment:
This can be simplified to mean the same thing, and also more similar to the other `if` clause for better readability.

```suggestion
    if (!isAddingRule || (widgetData.context === 'overlay' && widgetData.tokenId !== selectedTokenId)) {
```